### PR TITLE
feat(perf): enable config zstd

### PR DESCRIPTION
**Pull request by xkx9431**
##  Desciption
The server already supports ZSTD encoding; the client should adapt to this.

### Validation
image

### CODE REVIEW

**Comment by xkx9431** [comment]
@vicky-run @shoothzj  Please help review, thanks

File: opengemini/query_test.go [hunk 1]
```diff
@@ -77,7 +77,183 @@ func TestQueryWithMsgPack(t *testing.T) {
...
+		assert.Nil(t, err)
+		v, err := convertToInt64(result.Results[0].Series[0].Values[0][0])
+		if err != nil {
+			t.Fatalf("conversion error: %v", err)
+		}
+		assert.Equal(t, length, getTimestampLength(v))
+	}
+}
+
+// func TestQueryWithGZIPAndMsgPack(t *testing.T) {
```
Comment (xkx9431) [inline opengemini/query_test.go:190]: should enable when issue https://github.com/openGemini/opengemini-client-go/issues/200 fix

**Comment by xkx9431** [comment]
@xuthus5 @shoothzj  Please help review

File: opengemini/query_test.go (hunk 1 above)
Comment (xuthus5) [inline opengemini/query_test.go:190]: Remove commented out code.

File: opengemini/query.go [hunk 2]
```diff
@@ -112,17 +128,79 @@ func retrieveQueryResFromResp(resp *http.Response) (*QueryResult, error) {
...
+			return qr, err
 		}
+	default:
+		decompressedBody = body
 	}
-	return qr, nil
+
+	// Then, handle deserialization based on content type
+	switch contentType {
+	case "application/x-msgpack":
```
Comment (xuthus5) [inline opengemini/query.go:153]: Define them as constants?

**Review by xuthus5 — APPROVED** [review]
LGTM

**Pull request merged**